proxy-verifier/proxy-verifier-logic/src/handlers/mod.rs

```diff
@@ -0,0 +1,226 @@
+pub mod solidity_verifier_multi_part;
+pub mod solidity_verifier_standard_json;
+
+pub mod vyper_verifier_multi_part;
+pub mod vyper_verifier_standard_json;
+
+/************************************************/
+
+use crate::{
+    address_details, address_details::AddressDetails, to_hex::ToHex, Error, VerificationResponse,
+};
+use eth_bytecode_db_proto::blockscout::eth_bytecode_db::v2 as eth_bytecode_db_v2;
+use std::future::Future;
+
+async fn process_verification_request<'a, Request, RequestBuilder, Verify, VerifyOutput>(
+    eth_bytecode_db_client: &'a eth_bytecode_db_proto::http_client::Client,
+    contracts: Vec<(&blockscout_client::Client, ethers_core::types::Address)>,
+    request_builder: RequestBuilder,
+    verify: Verify,
+) -> VerificationResponse
+where
+    RequestBuilder: Fn(
+            ethers_core::types::Bytes,
+            eth_bytecode_db_v2::BytecodeType,
+            eth_bytecode_db_v2::VerificationMetadata,
+        ) -> Request
+        + Clone,
+    Verify: Fn(&'a eth_bytecode_db_proto::http_client::Client, Request) -> VerifyOutput + Clone,
+    VerifyOutput: Future<
+        Output = eth_bytecode_db_proto::http_client::Result<eth_bytecode_db_v2::VerifyResponse>,
+    >,
+{
+    let contract_details = retrieve_contract_details(&contracts).await;
+    if let Some(response) = check_invalid_contracts(&contract_details) {
+        return response;
+    }
+
+    let mut results = vec![];
+    for (contract_details, (blockscout_client, contract_address)) in
+        contract_details.into_iter().zip(contracts)
+    {
+        let result = verify_contract(
+            eth_bytecode_db_client,
+            contract_details,
+            request_builder.clone(),
+            verify.clone(),
+        )
+        .await;
+        match result {
+            Ok(()) => {
+                let search_result = search_contract(blockscout_client, contract_address).await;
```

Comment:
so, after so verified contract in `eth_bytecode_db` we will get already verified contract in blockscout? I thought that we have to go to some endpoint in blockscout like `search-contract-in-bytecode-db` or smthng

Reply:
or `blockscout_client::import::smart_contracts::get` performs searching in other sources?

Reply:
Yes, `blockscout_client::import::smart_contracts::get` is exactly `search-contract-in-bytecode-db` in that case